Libraries/LibWeb/TrustedTypes/TrustedTypePolicyFactory.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, Miguel Sacristán Izcue <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/TrustedTypes/TrustedTypePolicyFactory.h>
+
+#include <LibGC/Ptr.h>
+#include <LibJS/Runtime/Realm.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/DOM/QualifiedName.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/AttributeNames.h>
+#include <LibWeb/HTML/GlobalEventHandlers.h>
+#include <LibWeb/HTML/WindowEventHandlers.h>
+#include <LibWeb/Namespace.h>
+
+namespace Web::TrustedTypes {
+
+GC_DEFINE_ALLOCATOR(TrustedTypePolicyFactory);
+
+GC::Ref<TrustedTypePolicyFactory> TrustedTypePolicyFactory::create(JS::Realm& realm)
+{
+    return realm.create<TrustedTypePolicyFactory>(realm);
+}
+
+Optional<String> TrustedTypePolicyFactory::get_attribute_type(String& tagName, String& attribute, Optional<String> elementNS, Optional<String> attrNS)
+{
+    // 1. Set localName to tagName in ASCII lowercase.
+    auto const localName = tagName.to_ascii_lowercase();
+
+    // 2. Set attribute to attribute in ASCII lowercase.
+    attribute = attribute.to_ascii_lowercase();
+
+    // 3. If elementNs is null or an empty string, set elementNs to HTML namespace.
+    if (!elementNS.has_value() || elementNS.value().is_empty())
+        elementNS = String { Namespace::HTML };
+
+    // 4. If attrNs is an empty string, set attrNs to null.
+    if (attrNS.has_value() && attrNS.value().is_empty())
+        attrNS.clear();
+
+    // FIXME
+    // 5. Let interface be the element interface for localName and elementNs.
+    auto const& interface = localName;
```

Comment:
Ideally we'd have a method in `ElementFactrory` that could give us the element interface name for a given local name, but I don''t think we have anything currently that does that. 

For now, would something like this, which accounts for every row in the `get_trusted_type_data_for_attribute()` table work?

```cpp
String interface;
if (local_name == HTML::TagNames::iframe && element_ns == Namespace::HTML) {
    interface = "HTMLIFrameElement"_string;
} else if (local_name == HTML::TagNames::script && element_ns == Namespace::HTML) {
    interface = "HTMLScriptElement"_string;
} else if (local_name == SVG::TagNames::script && element_ns == Namespace::SVG) {
    interface = "SVGScriptElement"_string;
} else {
    interface = "Element"_string;
}
```

Reply:
I added your suggestion and kept a FIXME for the time that we actually have such a function.

This enabled the pass of 20+ more WPT tests. Thanks :)